packages/core-api/src/extensions/traversal.ts

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { isValidElement, ReactNode, ReactElement, Children } from 'react';
+
+export type Discoverer = (element: ReactElement) => ReactNode;
+
+export type Collector<Result, Context> = () => {
+  accumulator: Result;
+  visit(
```

Comment:
really a nitpick, but this one would be a good match for a single-object-parameter pattern

Reply:
Went for plain args as it's a bit of a tight loop and executed at initial render, so didn't want the extra wrapping. Main point of being that this is completely internal and totally agree that if this is gonna become a public interface we'd want to refactor :grin: